lib/test_prof/recipes/rspec/let_it_be.rb

```diff
@@ -114,6 +113,89 @@ def define_let_it_be_methods(identifier, **modifiers)
 
       let(identifier, &let_accessor)
     end
+
+    def define_freezing_hooks
+      # Prevent hooks from being defined several times
+      return if instance_variable_get(:"#{PREFIX}hooks_defined")
+
+      before(:all) do
+        let_it_be_objects = instance_variable_get(:"#{TestProf::LetItBe::PREFIX}let_it_be_objects")
+
+        let_it_be_objects.each { |object| Freezer.deep_freeze(object) }
+      end
+
+      prepend_after do |example|
+        if example.exception&.message&.match?(/can't modify frozen Hash/)
```

Comment:
Why Hash, btw?

What if we have:

```ruby
let_it_be(:users) { create_pair(:user) }

context "with one more user" do
  before { users << create(:user) }
end
```

(The example is very synthetic but anyway)

Reply:
We can also take a look at `exception.receiver`. For example, we can track all the let_it_be-frozen objects (e.g., by setting a _special_ instance variable or by using a WeakMap).

Reply:
Fair point on `create_pair`. `let_it_be` had no `reload` support for arrays when I touched leakage detection about a year ago, so it was always a `Hash`. I'll do a quick fix.

Reply:
> we can track all the let_it_be-frozen objects

That would make a huge difference to have a lead on the ~suspect~ victim at all times.